test/unittest/integration_tests/integration_test_fsu.cpp

```diff
@@ -0,0 +1,124 @@
+#include <optimizer.h>
+#include <sstream>
+#include <vector>
+
+using LayerHandle = std::shared_ptr<ml::train::Layer>;
+using ModelHandle = std::unique_ptr<ml::train::Model>;
+
+template <typename T>
+static std::string withKey(const std::string &key, const T &value) {
+  std::stringstream ss;
+  ss << key << "=" << value;
+  return ss.str();
+}
+
+template <typename T>
+static std::string withKey(const std::string &key,
+                           std::initializer_list<T> value) {
+  if (std::empty(value)) {
+    throw std::invalid_argument("empty data cannot be converted");
+  }
+
+  std::stringstream ss;
+  ss << key << "=";
+
+  auto iter = value.begin();
+  for (; iter != value.end() - 1; ++iter) {
+    ss << *iter << ',';
+  }
+  ss << *iter;
+
+  return ss.str();
+}
+
+TEST(fsu, simple_fc) {
+
+  std::unique_ptr<ml::train::Model> model = ml::train::createModel(
+    ml::train::ModelType::NEURAL_NET, {withKey("loss", "mse")});
+
+  model->addLayer(ml::train::createLayer(
+    "input", {withKey("name", "input0"), withKey("input_shape", "1:1:320")}));
+  model->addLayer(ml::train::createLayer(
+    "fully_connected",
+    {withKey("unit", 1000), withKey("weight_initializer", "xavier_uniform"),
+     withKey("bias_initializer", "zeros")}));
+  model->addLayer(ml::train::createLayer(
+    "fully_connected",
+    {withKey("unit", 1000), withKey("weight_initializer", "xavier_uniform"),
+     withKey("bias_initializer", "zeros")}));
+  model->addLayer(ml::train::createLayer(
+    "fully_connected",
+    {withKey("unit", 1000), withKey("weight_initializer", "xavier_uniform"),
+     withKey("bias_initializer", "zeros")}));
+  model->addLayer(ml::train::createLayer(
+    "fully_connected",
+    {withKey("unit", 1000), withKey("weight_initializer", "xavier_uniform"),
+     withKey("bias_initializer", "zeros")}));
+  model->addLayer(ml::train::createLayer(
+    "fully_connected",
+    {withKey("unit", 1000), withKey("weight_initializer", "xavier_uniform"),
+     withKey("bias_initializer", "zeros")}));
+  model->addLayer(ml::train::createLayer(
+    "fully_connected",
+    {withKey("unit", 100), withKey("weight_initializer", "xavier_uniform"),
+     withKey("bias_initializer", "zeros")}));
+
+  model->setProperty({withKey("batch_size", 1), withKey("epochs", 1),
+                      withKey("memory_swap", "true"),
+                      withKey("memory_swap_lookahead", "1"),
+                      withKey("model_tensor_type", "FP16-FP16")});
+
+  auto optimizer = ml::train::createOptimizer("sgd", {"learning_rate=0.001"});
+  model->setOptimizer(std::move(optimizer));
+
+  int status = model->compile(ml::train::ExecutionMode::INFERENCE);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+
+  status = model->initialize(ml::train::ExecutionMode::INFERENCE);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+
+  model->save("simplefc_weight_fp16_fp16_100.bin",
+              ml::train::ModelFormat::MODEL_FORMAT_BIN);
+  model->load("./simplefc_weight_fp16_fp16_100.bin");
```

Comment:
Simple question. Why do we need this `save` and `load` code? Is it because it only supports inference mode now?

Reply:
for test FSU, we need file that stored in storage. so save model file, and load files for now
--> in real Inference case, they have bin files already

Reply:
Oh yes, that's what the FSU is !🤣 Thank you for the kind reply :)